### DISCUSSION TITLE
How to set password please ？

### DISCUSSION THREAD

**Discussion by sunjianan9900** [ra1u/redis-dart#53] [Q&A] [answered]
How to set password please ？

**Comment by ra1u** [comment, marked as answer]
Hello. I guess you want to authenticate against redis server over secure connection

https://redis.io/commands/auth


``` Dart
final conn = RedisConnection();
Command command = await conn.connectSecure('host',port);
var rsp = await command.send_object(["AUTH",username,pazzword]);
print(rsp);
```
Let me know if that is what you are looking for.

Kind, regards Luka

**Reply by sunjianan9900** [reply]
thanks for your reply.
BTW
is not support set EXPlRE of key yet ?

**Reply by sunjianan9900** [reply]
send_object(["EXP","key", 30*24*60*60])
send_object(["EXPlRE","key", 30*24*60*60])
not work

**Reply by ra1u** [reply]
You can try with https://redis.io/commands/set
``` Dart
var rsp = await command.send_object(["SET", "anotherkey", "will expire in a minute","EX",60]);

```

**Reply by sunjianan9900** [reply]
thanks !!!

**Reply by ra1u** [reply]
Regarding EXPIRE https://redis.io/commands/expire

``` dart
send_object(["EXPlRE","key", 30246060])
```

 should work. (maybe it is that time needs to be string)
 However time you set is in seconds, so large numbers take long time to validate.

**Reply by ra1u** [reply]
I now noticed that Expire is not with i but lower case L. That might be an issue in your case.
